keps/159-scheduler-simulator-operator/README.md

```diff
@@ -0,0 +1,126 @@
+// Or, you can use an external etcd with EtcdURL.
+type SimulatorSpec struct {
+  // KubeAPIServerPort indicates the kube-apiserver’s port which the simulator has internally.
+  // (The each simulator has own kube-apiserver internally.)
+  // This field's value will be added to a "simulator" container's env as "KUBE_API_PORT".
+  KubeAPIServerPort            int 
+  // SimulatorServerPort indicates the port for the simulator’s endpoint.
+  // This field's value will be added to a "simulator" container's env as "PORT".
+  SimulatorServerPort          int 
+  // KubeSchedulerConfiguration indicates the configuration for the scheduler. 
+  // Simulators have the scheduler internally and you can configure it with this configuration.
+  // This field's value will be added to a "simulator" container's env as "KUBE_SCHEDULER_CONFIG_PATH".
+  // optional
+  KubeSchedulerConfiguration   string
+  // EtcdURL indicates the URL of etcd used in kube-apiserver.
+  // If this field is empty, the Pod created by this Simulator resource will have the etcd container.
+  // This field's value will be added to a "simulator" container's env as "KUBE_SCHEDULER_SIMULATOR_ETCD_URL".
+  // optional
+  EtcdURL *string
+  // CORSAllowedOriginList indicates the AllowedOriginList for CORS. 
+  // This is applied to kube-apiserver and simulator server.
+  // This field's value will be added to a "simulator" container's env as "CORS_ALLOWED_ORIGIN_LIST".
+  // optional.
+  CORSAllowedOriginList        []string
+  // Image should be the image of simulator.
+  // You can use your customized simulator here, but the simulator container created by this image should follow these things:
+  // - can be configured by the same environment variables. 
+  // - (will add other requirements if any)
+  Image                        string
+  // Describes the pod that will be created.
+  // This Pod should create the simulator.
+  //
+  // The system will add the container named "simulator" that: 
+  // - run the simulator created by image specified in Image field,
+  // - have Ports for kube-apiserver and the simulator endpoint,
+  // - have environment variables to configure the simulator.
+  // 
+  // And if the EtcdURL field is empty, the system will add the container named "etcd" for kube-apiserver.
+  // 
+  // You can overwrite those container spec by defining containers that have the same name, "simulator" or "etcd".
+  // 
+  // When PodTemplate has the container(s) named "simulator" or "etcd" and the field is different from the container spec generated by other fields in SimulatorSpec,
+  // container(s) definition in PodTemplate takes precedence.
+  PodTemplate *corev1.PodTemplateSpec 
+}
+
+type SimulatorStatus struct {
+  Phase SimulatorPhase
+
+  // A human readable message indicating details about why the simulator is in this phase.
+  // optional
+  Message *string 
+
+  // KubeAPIServerURL is the url to access the kube-apiserver in the simulator.
+  KubeAPIServerURL string
```

Comment:
Let's add `KubeConfig` or `KubeConfigPath`.

Reply:
As we discussed offline, KubeConfig isn't needed since users can communicate with the kube-apiserver in simulator only with url.